website/content/v1.3/kubernetes-guides/configuration/storage.md

```diff
@@ -107,6 +107,79 @@ talosctl -n <node ip> service kubelet restart
 
 Continue setting up [Mayastor](https://mayastor.gitbook.io/introduction/quickstart/deploy-mayastor) using the official documentation.
 
+### Piraeus / LINSTOR
+
+* [Piraeus-Operator](https://piraeus.io/)
+* [LINSTOR](https://linbit.com/drbd/)
+* [DRBD Extension](https://github.com/siderolabs/extensions#storage)
+
+<!-- markdownlint-disable MD007 -->
+<!-- markdownlint-disable MD032 -->
+<!-- markdownlint-disable MD025 -->
+
+{{< tabpane lang="yaml" >}}
+{{< tab header="Strategic merge patch" >}}
+machine:
+  install:
+    extensions:
+      - image: ghcr.io/siderolabs/drbd:v1.3.0
+  kernel:
+    modules:
+      - name: drbd
+      - name: drbd_transport_tcp
+{{< /tab >}}
+{{< tab header="JSON patch" >}}
+- op: add
+  path: /machine/install/extensions/-
+  value:
+    - image: ghcr.io/siderolabs/drbd:v1.3.0
+- op: add
+  path: /machine/kernel/modules/-
+  value:
+    - name: drbd
+    - name: drbd_transport_tcp
+{{< /tab >}}
+{{< /tabpane >}}
+
+```sh
+# Apply config update on node
+talosctl apply -f install-drbd.yml -n node01
+
+# Create dedicated namespace for piraeus
+kubectl create namespace piraeus
+
```

Comment:
# Add privileges to piraeus namespace
kubectl label ns piraeus pod-security.kubernetes.io/enforce=privileged
kubectl label ns piraeus pod-security.kubernetes.io/auth=privileged
kubectl label ns piraeus pod-security.kubernetes.io/warn=privileged

Reply:
Is it necessary to add these labels?
At least in my setup, it works without that.

Reply:
Hmm I believe it didn't work for me without it. I'll have to maybe try again on some vms otherwise yeah you can scrap it.

Reply:
With Talos 1.3 the security permissions are more strict, and Helm installs fails with:

```
0102 17:54:23.563110   18390 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "piraeus-operator" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "piraeus-operator" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "piraeus-operator" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "piraeus-operator" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

With the three label commands from above this error is gone, however I am in doubt if this is the best way to address this.

Reply:
I believe piraeus is requesting privileges for csi nodes and satellites. From there docs [](https://github.com/piraeusdatastore/piraeus-operator/blob/dcb28373adda761dbe76413690e8fbfc35d4ec28/doc/helm-values.adoc#pspprivilegedrole)

Reply:
Added commands to the instruction. Thanks for the hint!